plugins/main/bridge/bridge.go

```diff
@@ -293,12 +302,23 @@ func calcGatewayIP(ipn *net.IPNet) net.IP {
 }
 
 func setupBridge(n *NetConf) (*netlink.Bridge, *current.Interface, error) {
+	vlanFiltering := false
+	if n.Vlan != 0 {
+		vlanFiltering = true
+	}
 	// create bridge if necessary
-	br, err := ensureBridge(n.BrName, n.MTU, n.PromiscMode)
+	br, err := ensureBridge(n.BrName, n.MTU, n.PromiscMode, vlanFiltering)
 	if err != nil {
 		return nil, nil, fmt.Errorf("failed to create bridge %q: %v", n.BrName, err)
 	}
 
+	if n.Vlan != 0 {
+		err = netlink.BridgeVlanAdd(br, uint16(n.Vlan), false, true, true, false)
```

Comment:
I don't see this vlan filter entry being removed on `DEL`.  Are there any downsides to leaving it in place even after all veths tagged with the vid have been removed?

Reply:
Yeah I don't remove it from the bridge I can't find any good reason to remove it just link we are not removing the bridge if there is no veths connected to it.